gateway/handlers/createhandler.go

```diff
@@ -66,7 +75,8 @@ func MakeNewFunctionHandler(metricsOptions metrics.MetricOptions, c *client.Clie
 	}
 }
 
-func makeSpec(request *requests.CreateFunctionRequest, maxRestarts uint64, restartDelay time.Duration) swarm.ServiceSpec {
+func makeSpec(c *client.Client, request *requests.CreateFunctionRequest, maxRestarts uint64, restartDelay time.Duration) (swarm.ServiceSpec, error) {
```

Comment:
Perhaps move the secrets creation out of the spec creation and just pass it in, the spec creation is just an assembly. Makes it easier to test.

Reply:
Can do, I waffled on this one, originally writing it as you suggest, so it will be easy to switch it.